test_scripts/Protocol/commonProtocol.lua

```diff
@@ -198,4 +282,329 @@ function common.setProtectedServicesInIni()
+                end)
+            end, mobileConnectionDelay)
+        end)
+    end)
+end
+
+function common.delayedStartServiceAckP5(pDelayGetVT, pTS, pRpcServiceAckParams)
+  config.defaultProtocolVersion = 5
+
+  common.log("StartService")
+  common.startRpcService(pRpcServiceAckParams)
+  :ValidIf(function()
+      common.log("StartServiceAck")
+      if pTS.ts_get_si == nil then
+        return false, "StartServiceAck received before receiving of GetSystemInfo from HMI"
+      end
+      if pTS.ts_get_vt == nil and pDelayGetVT ~= -1 then
+        return false, "StartServiceAck received before receiving of GetVehicleType from HMI"
+      end
+      return true
+    end)
+  :Do(function()
+      common.log("RAI")
+      local ts_req = timestamp()
+      common.registerAppEx(common.vehicleTypeInfoParams.default)
+      :ValidIf(function()
+          local tolerance = 750
+          local ts_res = timestamp()
+          local act_delay = ts_res - ts_req
+          common.log("RAIResponse", act_delay)
+          if act_delay > tolerance then
+            return false, "RAI response is expected right after RAI request, actual delay: " ..
+            act_delay .. "ms"
+          end
+          return true
+        end)
+    end)
+end
+
+function common.delayedStartServiceAckP4(pDelayGetVT, pTS)
+  config.defaultProtocolVersion = 4
+
+  local ts_req = timestamp()
+  common.log("StartService")
+  common.getMobileSession():StartService(common.serviceType.RPC)
+  :Do(function()
+      common.log("RAI")
+      common.registerAppEx(common.vehicleTypeInfoParams.default)
+      :ValidIf(function()
+          common.log("RAIResponse")
+          if pTS.ts_get_si == nil then
+            return false, "RAI response received before receiving of GetSystemInfo from HMI"
+          end
+          if pTS.ts_get_vt == nil and pDelayGetVT ~= -1 then
+            return false, "RAI response received before receiving of GetVehicleType from HMI"
+          end
+          return true
+        end)
+    end)
+  :ValidIf(function()
+      local tolerance = 750
+      local ts_res = timestamp()
+      local act_delay = ts_res - ts_req
+      common.log("StartServiceAck", act_delay)
+      if act_delay > tolerance then
+        return false, "StartServiceAck is expected right after StartService request, actual delay: " ..
+        act_delay .. "ms"
+      end
+      return true
+   end)
+end
+
+function common.updateHMICapabilitiesFile(pVehicleTypeData)
+    local hmiCapTbl = common.getHMICapabilitiesFromFile()
+    hmiCapTbl.VehicleInfo.vehicleType.make = pVehicleTypeData.make
+    hmiCapTbl.VehicleInfo.vehicleType.model = pVehicleTypeData.model
+    hmiCapTbl.VehicleInfo.vehicleType.modelYear = pVehicleTypeData.modelYear
+    hmiCapTbl.VehicleInfo.vehicleType.trim = pVehicleTypeData.trim
+    common.setHMICapabilitiesToFile(hmiCapTbl)
+end
+
+function common.getRpcServiceAckParamsFromStruct(pVehicleTypeInfoParams)
```

Comment:
I would combine this function with `getRpcServiceAckParams` since they are pretty similar

Reply:
Done in 40ee931